src/inmanta/server/services/resourceservice.py

```diff
@@ -390,6 +358,53 @@ def in_requires(req: ResourceIdStr) -> bool:
 
         return 200, {"environment": env.id, "agent": agent, "version": version, "resources": deploy_model}
 
+    async def _mark_deployed(
+        self,
+        env: data.Environment,
+        neg_increment: abc.Set[ResourceIdStr],
+        now: datetime.datetime,
+        on_agent: Callable[[ResourceIdStr], bool],
```

Comment:
I would also make this optional. It feel rather strange to have to pass in a lambda function that always returns True when you don't want to apply a filter.

Reply:
I would not make it optional then, but default to the lambda that returns true

Reply:
> I would not make it optional then, but default to the lambda that returns true

That's exactly what I meant. I was referring to the fact that passing the attribute should be optional and you to whether or not the type should be optional :p